lock.rkt

```diff
@@ -0,0 +1,40 @@
+#lang racket/base
+
+;; A Readers Writer Lock
+(struct RWLock
+  (access-sema write-sema readers)
+  #:mutable)
```

Comment:
Maybe the file name should be `rwlock.rkt`, and move into `base/` directory (create one), since the functionality is not directly related to the server.